BitwardenShared/UI/Platform/Application/AppProcessorTests.swift

```diff
@@ -200,6 +200,46 @@ class AppProcessorTests: BitwardenTestCase { // swiftlint:disable:this type_body
         XCTAssertIdentical(syncService.delegate, subject)
     }
 
+    /// `handleAppLinks(URL)` navigates the user based on the input URL.
+    func test_init_handleAppLinks() {
+        let url = URL(string:
+            "https://bitwarden.com/finish-signup?email=example@email.com&token=verificationtoken&fromEmail=true"
+        )
+        subject.handleAppLinks(incomingURL: url!)
+
+        XCTAssertEqual(coordinator.routes.last, .auth(.completeRegistrationFromAppLink(
+            emailVerificationToken: "verificationtoken",
+            userEmail: "example@email.com",
+            fromEmail: true,
+            region: .unitedStates
+        )))
+    }
+
+    /// `handleAppLinks(URL)` checks error report for `.appLinksInvalidURL`.
+    func test_init_handleAppLinks_invalidURL() {
+        let url = URL(string: "https://bitwarden.com/")
+        subject.handleAppLinks(incomingURL: url!)
+        XCTAssertEqual(errorReporter.errors.last as? AppProcessorError, .appLinksInvalidURL)
+    }
+
+    /// `handleAppLinks(URL)` checks error report for `.appLinksInvalidPath`.
+    func test_init_handleAppLinks_invalidPath() {
+        let url = URL(
+            string: "https://bitwarden.com/not-valid?email=example@email.com&token=verificationtoken&fromEmail=true"
+        )
+        subject.handleAppLinks(incomingURL: url!)
+        XCTAssertEqual(errorReporter.errors.last as? AppProcessorError, .appLinksInvalidPath)
+    }
+
+    /// `handleAppLinks(URL)` checks error report for `.appLinksInvalidParametersForPath`.
+    func test_init_handleAppLinks_invalidParametersForPath() {
```

Comment:
🎨 This is only covering the `email` existing and the other ones not. But IMO we at least should add 2 more tests when `verificationToken` exists and not the others and finally when `fromEmail` exists but not the others.